lib/client/doc.js

```diff
@@ -919,6 +919,17 @@ Doc.prototype.resume = function() {
   this.flush();
 };
 
+// Create a snapshot that can be serialized, deserialized, and passed into `Doc.ingestSnapshot`.
+//
+// Note that types.defaultType is used inside ingestSnapshot when snapshot.type is omitted.
+// If you need to use a different type, populate `snapshot.type` with the OT type you are using.
+Doc.prototype.createSnapshot = function() {
```

Comment:
Since `createSnapshot` sounds like it might have a side effect, how about `asSnapshot` or `toSnapshot`?

Reply:
Renamed.